Sources/PT.PM.Matching/Patterns/PatternAny.cs

```diff
@@ -1,19 +1,71 @@
-﻿using PT.PM.Common.Nodes;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using PT.PM.Common;
+using PT.PM.Common.Nodes;
+using PT.PM.Common.Nodes.Tokens.Literals;
 
 namespace PT.PM.Matching.Patterns
 {
-    public class PatternAny : PatternUst<Ust>
+    public class PatternAny : PatternUst<Ust>, IRegexPattern
     {
         public static PatternAny Instance = new PatternAny();
 
+        public string Default => ".*";
+
+        public string RegexString
+        {
+            get => Regex.ToString();
+            set => Regex = new Regex(string.IsNullOrEmpty(value) ? Default : value, RegexOptions.Compiled);
+        }
+
+        public Regex Regex { get; private set; }
+
+        public override bool Any => Regex == null || RegexString.Equals(Default);
+
+        public PatternAny()
+        {
+        }
+
+        public PatternAny(string regex)
+        {
+            RegexString = regex;
+        }
+
         public override MatchContext Match(Ust ust, MatchContext context)
         {
-            if (ust == null)
+            if (Any)
             {
-                return context.MakeSuccess();
+                if (ust == null)
+                {
+                    return context.MakeSuccess();
+                }
+                return context.AddMatch(ust);
             }
 
-            return context.AddMatch(ust);
+            if (ust.Children.Length > 0)
```

Comment:
Why did you write this line? It's totally incorrect and makes PatternAny with regex senseless.